src/plugin/protocol-smb/ivy.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" ?>
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+
+<ivy-module xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+  xsi:noNamespaceSchemaLocation="http://ant.apache.org/ivy/schemas/ivy.xsd" 
+  xmlns:ns0="http://ant.apache.org/ivy/maven" version="2.0">
+  <info organisation="org.apache.nutch" module="${ant.project.name}">
+    <license name="Apache 2.0"/>
+    <ivyauthor name="Apache Nutch Team" url="http://nutch.apache.org"/>
+    <description>
+        Apache Nutch
+    </description>
+  </info>
+
+  <configurations>
+    <include file="../../..//ivy/ivy-configurations.xml"/>
+  </configurations>
+
+  <publications>
+    <!--get the artifact from our module name-->
+    <artifact conf="master"/>
+  </publications>
+
+  <dependencies>
+    <dependency org="com.hierynomus" name="smbj" rev="0.13.0"/>
+    <dependency org="net.engio" name="mbassador" rev="1.3.0"/>
+    <dependency org="org.bouncycastle" name="bcprov-jdk18on" rev="1.75"/>
+    <dependency org="com.hierynomus" name="asn-one" rev="0.6.0"/>
+    <dependency org="commons-io" name="commons-io" rev="2.17.0"/>
```

Comment:
`commons-io` is already defined in `ivy/ivy.xml` (albeit an older version). It looks like you are only using this dependency for the `IOUtils.toByteArray` method so I wonder we can rework that and remove the dependency?
What about the methods in [DeflateUtils.java](https://github.com/apache/nutch/blob/master/src/java/org/apache/nutch/util/DeflateUtils.java). Would any of those work?